### PR TITLE
fix: Cohere namespace reorg

### PR DESCRIPTION
**Pull request by vblagoje**
- Reorganize namespace for generators, put both generators under cohere
- fixes https://github.com/deepset-ai/haystack/issues/6824
- Fixes for pre-release checks
  - make sure all examples streaming and non streaming, component alone and in pipeline work well (both generators)

### CODE REVIEW

**Comment by vblagoje** [comment]
@masci this PR includes minor fixes I discovered during pre-release testing and doc sync. I also fixed https://github.com/deepset-ai/haystack/issues/6824 @annthurium found yesterday.

**Comment by annthurium** [comment]
yay! Thanks for the fix 😸

**Comment by annthurium** [comment]
do we need to give @dfokina a heads up to change the docs for the `CohereGenerator` ?

File: integrations/cohere/src/haystack_integrations/components/generators/cohere/chat/__init__.py [hunk 1]
```diff
@@ -1,6 +0,0 @@
-# SPDX-FileCopyrightText: 2023-present deepset GmbH <info@deepset.ai>
-#
-# SPDX-License-Identifier: Apache-2.0
```
Comment (masci) [inline integrations/cohere/src/haystack_integrations/components/generators/cohere/chat/__init__.py:3]: let's keep these 3 lines

**Review by masci — DISMISSED** [review]
Let's retain the license header and it's good to merge

**Review by julian-risch — APPROVED** [review]
LGTM! 👍

**Review by silvanocerza — APPROVED** [review]

**Pull request merged**